portal/templates/initial_queries_macros.html

```diff
@@ -8,14 +9,26 @@ <h4 class="tnth-headline">{{ _("Terms of Use") }}</h4>
             <div id="termsText">
               {{ terms['asset']|safe if terms}}
             </div>
-            <input type="hidden" id="termsURL" data-url="{{ terms['agreement_url']|safe if terms}}" value=""/>
             <div id="termsExpand" class="tnth-hide">
               <a href=""><i class="fa fa-plus"></i> {{ _("View TrueNTH Terms") }}</a>
             </div>
             <div id="termsCheckbox">
-              <label id="agreeLabel" data-agree="false" data-type="terms">
-                <i class="fa fa-square-o fa-lg"></i> <span>{{ _("I consent and agree to the above terms.") }}</span>
-              </label>
+               <div class="terms-checkbox-container">
+               <!-- note these checkboxes are selectively hidden using css, as some are applicable only in Truenth and some only applicable in Eproms-->
+                <label class="custom-tou tnth-hide" data-agree="false" data-type="terms" data-tou-type="subject website consent" data-core-data-type="subject_website_consent" data-url="{{terms['url'] if terms}}">
+                  <i class="fa fa-square-o fa-lg" class="edit-view"></i> <span class="edit-view">{{ _("I have read the information notice above and consent and agree to the processing of my personal information (including my health information) on the terms described in this Consent and Terms and Conditions.") }}</span>
+                  <div class="display-view text-warning tnth-hide">{{_("You have previously provided your consent to the website during a visit at your treating site. If you would like a copy of this please contact your study contact.")}}<br/></div>
+                </label>
+                <br/>
+                <label class="general-tou" data-agree="false" data-type="terms" data-tou-type="website terms of use" data-core-data-type="website_terms_of_use" data-url="{{terms['url'] if terms}}">
+                    <i class="fa fa-square-o fa-lg"></i>&nbsp;<span class="default-tou-text">{{ _("I consent and agree to the above terms.") }}</span>
+                    <span class="custom-tou-text">{{ _("I have read the website ")}}<a href="{{url_for('portal.privacy')}}" target="_blank" class="required-link" data-core-data-subtype="privacy_policy" data-tou-type="privacy policy">{{_('privacy policy')}}</a>{{_(' and ')}}<a href="{{url_for('portal.terms_and_conditions')}}" target="_blank" class="required-link">{{_("terms") }}</a></span>
```

Comment:
Unfortunately it's nearly impossible to translate content concatenated in this way. You'll have to do some of the most disgusting string formatting imaginable, unless you know of a better way (I would *love* to find one). It's fine to include HTML in translation calls when necessary but since these links are dynamically generated it requires all that gross string formatting.
```
<span class="custom-tou-text">
{{ _(
    "I have read the website %(privacy_link)sprivacy policy%(link_end)s and %(terms_link)sterms%(link_end)s",
    privacy_link=('<a href="%s" target="_blank" class="required-link" data-core-data-subtype="privacy_policy" data-tou-type="privacy policy">' % url_for('portal.privacy')), 
    terms_link=('<a href="%s" target="_blank" class="required-link">' % url_for('portal.terms_and_conditions')),
    link_end='</a>'
) }}
</span>

```
Looks like the [`blocktrans` Jinja tag](https://docs.djangoproject.com/en/1.7/topics/i18n/translation/#blocktrans-template-tag) might be a little better:
```
{% blocktrans with privacy_url=url_for('portal.privacy') terms_url=url_for('portal.terms_and_conditions') %}
I have read the website <a href="{{privacy_url}}" target="_blank" class="required-link" data-core-data-subtype="privacy_policy" data-tou-type="privacy policy">privacy policy</a> and <a href="{{terms_url}}" target="_blank" class="required-link">terms</a>
{% endblocktrans %}
```

In any case, I think `url_for()` will need to be rendered before passing to the template